kolibri/core/assets/src/views/connection-snackbars/index.vue

```diff
@@ -49,22 +53,51 @@
     },
     computed: {
       disconnected() {
-        return this.currentSnackbar === ConnectionSnackbars.DISCONNECTED;
+        return !this.connected && this.currentSnackbar === ConnectionSnackbars.DISCONNECTED;
       },
       tryingToReconnect() {
-        return this.currentSnackbar === ConnectionSnackbars.TRYING_TO_RECONNECT;
+        return !this.connected && this.currentSnackbar === ConnectionSnackbars.TRYING_TO_RECONNECT;
       },
       successfullyReconnected() {
-        return this.currentSnackbar === ConnectionSnackbars.SUCCESSFULLY_RECONNECTED;
+        return (
+          this.connected && this.currentSnackbar === ConnectionSnackbars.SUCCESSFULLY_RECONNECTED
+        );
       },
       remainingTime() {
-        return new Date(1000 * this.reconnectTime).toISOString().substr(14, 5);
+        return new Date(1000 * this.timeToReconnect).toISOString().substr(14, 5);
+      },
+    },
+    watch: {
+      currentSnackbar: 'setTimer',
+      connected: 'setTimer',
+    },
+    mounted() {
+      this.setTimer();
+    },
+    beforeDestroy() {
+      this.clearTimer();
+    },
+    methods: {
+      clearTimer() {
+        if (this.timer !== null) {
+          clearInterval(this.timer);
+          this.timer = null;
+        }
+      },
+      setTimer() {
+        this.clearTimer();
+        if (this.reconnectTime) {
+          this.timeToReconnect = this.reconnectTime;
+          this.timer = setInterval(() => {
+            this.timeToReconnect = this.timeToReconnect - 1;
```

Comment:
Can use assignment operator `this.timeToReconnect -= 1;`

Reply:
I'm not a fan of it, personally. I think we used to have this set to never previously: https://eslint.org/docs/rules/operator-assignment